test/main.hs

```diff
@@ -13,6 +13,12 @@ import System.IO (hClose, openBinaryTempFile, hGetContents)
 import qualified Data.ByteString as S
 import qualified Data.ByteString.Char8 as S8
 import System.Directory (getTemporaryDirectory, removeFile)
+import System.Info (os)
+
+ifWindows :: IO () -> IO ()
+ifWindows action
+  | os != "windows" = return ()
```

Comment:
```suggestion
  | os /= "windows" = return ()
```

Though the logic overall may be easier to read with:

```haskell
ifWindows action
  | os == "windows" = action
  | otherwise = pure ()
```